pkgs/top-level/perl-packages.nix

```diff
@@ -6731,6 +6758,48 @@ let
     };
   };
 
+  ham = buildPerlPackage rec {
+    date = "20180212";
+    rev = "9c5a38941d541516c7c9617de5c8b59db5bc1717";
+    name = "ham-${date}." + (stdenv.lib.substring 0 8 "${rev}");
```

Comment:
Nixpkgs likes to format `version` when it's a commit from a repo as `YYYY-MM-DD` being the timestamp of the commit.
Also add `unstable` to the name.

> https://nixos.org/nixpkgs/manual/#sec-package-naming